### DISCUSSION TITLE
Using Different API Versions in a module

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3373] [General] [answered]
Hi,

How do I use different API versions in a module.

For example, 

1) Module-1 using API version v1.
2) Module-2 using API version v2. Latest API version is v3.
3) In Module-3, we need to consume Module-1 API v1 & Module-2 API v2.

**Comment by sbwalker** [comment, marked as answer]
@Sharafudeen you would need to include an API versioning strategy in your code - ie. include the version# in the Url as a segment or querystring parameter... or in the request header sent by the client. Generally I always try to make my APIs backward compatible so that I do not need to deal with API versioning explicitly in my code - but this is a choice, not a restriction.
